job-runner/chain/redis_repo.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2017, Square, Inc.
+
+package chain
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/garyburd/redigo/redis"
+)
+
+// ChainKey is the keyspace for serialized Chains
+const CHAIN_KEY = "ChainById"
+
+var (
+	ErrKeyExists           = errors.New("key already exists in redis")
+	ErrKeyNotFound         = errors.New("key not found in redis")
+	ErrMultipleKeysDeleted = errors.New("multiple keys deleted from redis")
+)
+
+// A RedisConnectionPool is used for pooling redis connections.
+type RedisConnectionPool interface {
```

Comment:
So far I don't think this needs to be exported/public. Would the creator (probably just main.go) need to provide this? That seems too much to ask callers. More likely, caller will create a Redis repo with an arg like `maxConnections` or `poolSize` or something, then leave it to the Redis repo itself to handle the internal details. (Looking below, I see `RedisRepoConfig` could be used for this.)